### ISSUE TITLE
mail.py should conform to docstring and whitespace standards

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6649]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)| @staceysern reported|
|-|-|
|Trac ID|trac#6649|
|Type|enhancement|
|Created|2013-07-28 03:56:21Z|
|Branch|https://github.com/twisted/twisted/tree/mail-doc-6649|

The docstrings in mail.py should be updated to provide descriptions of all classes, functions and methods as well as their parameters and attributes. The spacing before classes and functions should conform to the coding standards.

<details><summary>Searchable metadata</summary>

```
trac-id__6649 6649
type__enhancement enhancement
reporter__shira shira
priority__normal normal
milestone__ 
branch__branches_mail_doc_6649 branches/mail-doc-6649
branch_author__shira shira
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__documentation documentation
time__1374983781000000 1374983781000000
changetime__1378756996000000 1378756996000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_1"></a>|@staceysern commented|
|-|-|

(In [39350]) Branching to 'mail-doc-6649'

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_2"></a>|@staceysern commented|
|-|-|

(In [39387]) Fix docstrings and whitespace.

refs [#6649](https://github.com/twisted/twisted/issues/6649)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)|@staceysern removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_4"></a>|@staceysern commented|
|-|-|

(In [39395]) Fix formatting.

refs [#6649](https://github.com/twisted/twisted/issues/6649)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_5"></a>|@staceysern commented|
|-|-|

(In [39406]) Modified IAliasableDomain.exists docstring.

refs [#6649](https://github.com/twisted/twisted/issues/6649)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)|@staceysern set owner to @staceysern|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_7"></a>|@staceysern commented|
|-|-|

(In [39776]) Revised API documentation

refs [#6649](https://github.com/twisted/twisted/issues/6649)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_8"></a>|@staceysern commented|
|-|-|

(In [39777]) Remove blank lines after class docstrings

refs [#6649](https://github.com/twisted/twisted/issues/6649)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_9"></a>|@staceysern commented|
|-|-|

(In [39780]) Going back to original code after discovering twistedchecker problem

refs [#6649](https://github.com/twisted/twisted/issues/6649)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_10"></a>|@staceysern removed owner|
|-|-|

Revised to incorporate comments from other mail documentation code reviews and for general improvement in readability. Revisions include:

 * Changing type references from str to bytes 
 * Using L{} instead of C{} for standard library types 
 * Including a locally usable name in L{} 
 * Removing blank lines after class docstrings 
 * Removing (optional) in param descriptions 
 * Changing the format of type descriptions of tuples 
 * Changing the top file message

This builds with twistedchecker error 9016 on line 376 due to a bug in twistedchecker (bug report #1217379).

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_12"></a>|@lvh removed owner|
|-|-|

Hi shira,

Thanks for working on this. Documentation in crufty old bits of Twisted is often what turns people off from Twisted :)

1.  twisted/mail/mail.py  L60:  +        if the name doesn't exists in this dictionary. 
2.  twisted/mail/mail.py  L98:  +        if the name doesn't exists in this dictionary. 

Everything else looks good! Please merge after these two tidbits.

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_13"></a>|@staceysern commented|
|-|-|

(In [39898]) Fix typos noted in code review

refs [#6649](https://github.com/twisted/twisted/issues/6649)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_14"></a>|@staceysern set status to `closed`|
|-|-|

(In [39923]) Merge mail-doc-6649: Fix docstrings and whitespace for mail.py

Author: shira
Reviewer: lvh
Fixes: [#6649](https://github.com/twisted/twisted/issues/6649)
